src/IniReader.php

```diff
@@ -284,7 +284,15 @@ private function readWithAlternativeImplementation($ini)
             }
 
             if (is_string($value)) {
-                $value = trim($value, "'\"");
+                if (preg_match('/^"(.*)"$/', $value)) {
+                    $value = preg_replace('/^"(.*)"$/', '$1', $value);
+                    $value = str_replace('\"', '"', $value);
+                } elseif (preg_match("/^'(.*)'$/", $value)) {
+                    $value = preg_replace("/^'(.*)'$/", '$1', $value);
+                    $value = str_replace("\'", "'", $value);
+                } else {
+                    $value = trim($value, "'\"");
+                }
```

Comment:
This actually fixes some difference between this reader implementation and `parse_ini_string`.
Values like `key = "test \" test"` or `key = "' test"` resulted in different results. This should be covered by the additional tests